### PR TITLE
Add documentation for adding a new benchmark

### PR DESCRIPTION
**Pull request by Wal-eed**
Updated documentation to provide concise details on how to add new benchmarks to seL4bench. Code comments were added to highlight places in the code that require modification when adding a new benchmark

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,3 +73,22 @@ hardware tests.
 Since this benchmark will cause the kernel image to be an EL2 image, it
 will have an impact on the observed numbers for the other benchmark
 applications as well, since they'll be using an unexpected kernel build.
+
+
+# Adding a new benchmark
+Contributing a new benchmark to seL4bench requires a few steps:
+* Under `apps`, create a directory for your new benchmark and:
```
Comment (lsf37) [inline README.md:80]: After titles and before lists there should be a blank line in markdown (GitHub will still process it correctly, but other markdown engines don't necessarily)

```suggestion
# Adding a new benchmark

Contributing a new benchmark to seL4bench requires a few steps:

* Under `apps`, create a directory for your new benchmark and:
```

File: README.md [hunk 2]
```diff
@@ -73,3 +73,22 @@ hardware tests.
...
+    * Provide a CMakelists.txt file that defines a new executable.
```
Comment (lsf37) [inline README.md:81]:
```suggestion
    * Provide a `CMakelists.txt` file that defines a new executable.
```

File: README.md [hunk 3]
```diff
@@ -73,3 +73,22 @@ hardware tests.
...
+    * Provide a `src` folder that contains the source code for your
+      benchmark.
+* Under `apps/sel4bench`:
+    * Update CMakeLists.txt to add your new benchmark to the list of
```
Comment (lsf37) [inline README.md:85]:
```suggestion
    * Update `CMakeLists.txt` to add your new benchmark to the list of
```

File: README.md [hunk 4]
```diff
@@ -73,3 +73,22 @@ hardware tests.
...
+      benchmarks.
+    * Under `src`:
+        * Update `benchmark.h` to include your generated config for your benchmark, as well as provide a function declaration that will act as the entry point for your benchmark.
```
Comment (lsf37) [inline README.md:88]:
```suggestion
        * Update `benchmark.h` to include your generated config for your benchmark, and provide a function declaration that will act as the entry point for your benchmark.
```

**Review by lsf37 — APPROVED** [review]
Thanks, that is a nice idea!

Please rephrase the git commit to have a title < 50 characters and a body explaining the reason/intention for the change.

I've left suggestions for minor tweaks in the text/layout.

**Comment by Wal-eed** [comment]
> Thanks, that is a nice idea!
> 
> Please rephrase the git commit to have a title < 50 characters and a body explaining the reason/intention for the change.
> 
> I've left suggestions for minor tweaks in the text/layout.

Thanks, will update with the minor tweaks

**Pull request merged**